### ISSUE TITLE
Why set the first element of Attention'output to zero? (line 172 of tratransformer_flow.py)

### ISSUE THREAD

**Issue by Minokori** [apple/ml-tarflow#8]
In `transformer_flow.py`, class `MetaBlock`'s `forward()` method, line 172, I noticed that the first element (The matrix index is `[:,0,:]`) of Attention'output was set to zeros:
```python
x = torch.cat([torch.zeros_like(x[:, :1]), x[:, :-1]], dim=1)
# zeor_like  -> (B, 1,   C_out)
# x[:,:-1]   -> (B, T-1, C_out)
# concat     -> (B, T,   C_out)
```
In the original paper, this element corresponds to $z_0 = x_0$ described in equation (3), when $i = 0$ 
(I omitted the superscript in order to focus on discussing what happens when calculating $z_i$ from input $x_i$ before concatenating all $z_i$ outputting $z^{t+1}$.)

Can I understand that, due to $x_{<i}$ being empty when $i=0$, according to the subexpression for $i>0$ in equation (3), $\mu_{i}$ and $\alpha_{i}$ have no effect in the equation, leading to it degenerating into the special form for $i=0$, resulting in the first element and the parameters of Attention being gradient-independent, thus using zero padding to avoid interference during training?

If possible, I hope you can answer my question.

**Comment by Shuangfei** [comment]
Your understanding is correct. The first position z_0 goes through an identity transformation and zero padding is an easy way of doing it.

**Comment by Minokori** [comment]
Thank you for your prompt reply!
